ui/packages/atlasmap/src/Atlasmap/utils/field.ts

```diff
@@ -85,100 +85,126 @@ export function getConstantTypeIndex(constVal: string): number {
-  let field = cfg.propertyDoc.getField(propName);
+  let field = isSource
+    ? cfg.sourcePropertyDoc.getField(propName)
+    : cfg.targetPropertyDoc.getField(propName);
   if (!field) {
     field = new Field();
   }
   field.name = propName;
-  field.value = propValue;
   field.type = propType;
   field.scope = propScope;
-  field.docDef = cfg.propertyDoc;
   field.userCreated = true;
-  cfg.propertyDoc.addField(field);
+
+  if (isSource) {
+    field.docDef = cfg.sourcePropertyDoc;
+    cfg.sourcePropertyDoc.addField(field);
+  } else {
+    field.docDef = cfg.targetPropertyDoc;
+    cfg.targetPropertyDoc.addField(field);
+  }
   cfg.mappingService.notifyMappingUpdated();
 }
 
-export function deleteProperty(propName: string): void {
+export function deleteProperty(propName: string, isSource: boolean): void {
   const cfg = ConfigModel.getConfig();
-  const field = cfg.propertyDoc.getField(
-    cfg.propertyDoc.pathSeparator + propName.split(" ")[0],
-  );
+  const field = isSource
+    ? cfg.sourcePropertyDoc.getField(
+        cfg.sourcePropertyDoc.pathSeparator + propName.split(" ")[0],
+      )
+    : cfg.targetPropertyDoc.getField(
+        cfg.sourcePropertyDoc.pathSeparator + propName.split(" ")[0],
+      );
   if (!field) {
     return;
   }
   cfg.mappingService.removeFieldFromAllMappings(field);
-  cfg.propertyDoc.removeField(field);
+  if (isSource) {
+    cfg.sourcePropertyDoc.removeField(field);
+  } else {
+    cfg.targetPropertyDoc.removeField(field);
+  }
   cfg.mappingService.notifyMappingUpdated();
 }
 
 export function editProperty(
   propName: string,
-  propValue: string,
   propType: string,
   propScope: string,
   newName: string,
+  isSource: boolean,
 ): void {
   const cfg = ConfigModel.getConfig();
-  let field = cfg.propertyDoc.getField(
-    cfg.propertyDoc.pathSeparator + propName,
-  );
+  let field = isSource
+    ? cfg.sourcePropertyDoc.getField(
+        cfg.sourcePropertyDoc.pathSeparator + propName,
+      )
+    : cfg.targetPropertyDoc.getField(
+        cfg.targetPropertyDoc.pathSeparator + propName,
+      );
   if (!field) {
     return;
   }
   if (propName !== newName) {
     field.name = newName;
   }
-  field.value = propValue;
   field.type = propType;
   field.scope = propScope;
-  cfg.propertyDoc.updateField(field, "");
-  cfg.mappingService.notifyMappingUpdated();
-}
-
-export function getPropertyValue(propName: string): string {
-  const cfg = ConfigModel.getConfig();
-  const field = cfg.propertyDoc.getField(
-    cfg.propertyDoc.pathSeparator + propName.split(" ")[0],
-  );
-  if (!field) {
-    return "";
+  if (isSource) {
+    cfg.sourcePropertyDoc.updateField(field, "");
+  } else {
+    cfg.targetPropertyDoc.updateField(field, "");
   }
-  return field.value;
+  cfg.mappingService.notifyMappingUpdated();
 }
 
-export function getPropertyType(propName: string): string {
+export function getPropertyType(propName: string, isSource: boolean): string {
```

Comment:
other benefit of passing in the document as a function argument is that you don't have to access the config object I guess.